### PR TITLE
Update AGP to v.8.3.1

### PR DESCRIPTION
**Pull request by siddh1004**

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -4,7 +4,11 @@
 
 ### Internal
 
-- Bump AGP to v8.3.0
+- Bump AGP to v8.3.1
+- Bump kotlin to v1.9.23
+- Bump Compose compiler to v1.5.11
+- Bump ksp to v1.9.23-1.0.19
+- Bump Android desugar JDK library to v2.0.4
```
Comment (msasikanth) [inline CHANGELOG.md:11]: Can we separate them into separate PRs? You can combine Kotlin, KSP and Compose dependencies into a single PR.

Desugar JDK dependency update is in this PR
https://github.com/simpledotorg/simple-android/pull/4843

**Review by msasikanth — CHANGES_REQUESTED** [review]

File: CHANGELOG.md (hunk 1 above)
Comment (siddh1004) [reply CHANGELOG.md:11]: @msasikanth Kotlin, KSP and Compose dependencies are dependent to AGP update. I will remove the Desugar JDK dependency update as there is already a separate PR for that.

Comment (msasikanth) [reply CHANGELOG.md:11]: We can update Kotlin, KSP and Compose once AGP PR is merged it. Let that branch depending on the AGP update branch.

Comment (siddh1004) [reply CHANGELOG.md:11]: @msasikanth It would fail the build

Comment (msasikanth) [reply CHANGELOG.md]: The AGP update will fail?

Comment (msasikanth) [reply CHANGELOG.md]: As far as I know, AGP updates shouldn't depend on Kotlin updates. Let me check that locally once and get back

Comment (msasikanth) [reply CHANGELOG.md]: I am not seeing any dependencies on Kotlin, KSP or Compose when updating AGP. I am able to build the project and run it on device.

Comment (siddh1004) [reply CHANGELOG.md]: I have pushed the change with only AGP update

**Review by msasikanth — APPROVED** [review]
